src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -2717,14 +2718,18 @@ public enum RisingRivals implements LogicCardInfo {
         return basic (this, hp:HP050, type:WATER, retreatCost:1) {
           weakness L, PLUS10
           move "Cosmic Draw", {
-            text "If your opponent has any Evolved Pokémon in play, draw 3 cards."
+            text "If your opponent has any Evolution Pokémon in play, draw 3 cards."
+            //Errata: Evolution 「進化ポケモン」, not Evolved 進化している. https://www.pokemon-card.com/card-search/details.php/card/22091/regu/all
             energyCost C
             attackRequirement {
-              assert my.deck : "Your deck is empty"
-              assert opp.all.find{it.evolution} : "Your opponent doesn't have any Evolved Pokémon in play"
+              assert my.deck : "Your deck is empty"              
```

Comment:
why did we remove the condition here?

Reply:
It moved into onAttack, so the attack can still be used if the opponent doesn't have any Evolutions in play, but it will just do nothing.

Side note, compulsory switch Drag Off should also work like this, but that's a problem for future me to solve.

Reply:
but why should we allow an attack to do nothing? in case that's a misplay, isn't that bad for the user?

Reply:
True, but that's the way it works in the physical TCG. If your opponent misplays and tries to attack even though it will do nothing, you are within your rights to not allow them to take it back. Unlike Trainers, players can use attacks even though those attacks will have no effect.

One card that does make a difference between an attack that can't be used vs an attack that will do nothing is Holon Circle: if you're trying to break a Holon Circle with a Staryu that has a non-Water Energy attached, then having a Cosmic Draw that does nothing will let you do so (you declare the attack, then Holon Circle triggers and gets discarded), but having a Cosmic Draw that has an attack requirement will not (because you can't even declare the attack to begin with).

That said, I am OK with it staying as an attack requirement for user-friendliness. It's unlikely to matter much either way since neither Staryu nor Holon Circle are commonly played cards, so for it to manifest, not only would you need two players with rarely played cards in their decks matching against each other, the Staryu player would also need to be in a situation where he has a non-Water Energy attached, which puts it further in the realm of impossibility.

Reply:
Great, do you want me to make the change?

Reply:
Sure